controllers/storagecluster/volumegroupsnapshotterclasses.go

```diff
@@ -0,0 +1,173 @@
+package storagecluster
+
+import (
+	"fmt"
+	"reflect"
+
+	groupsnapapi "github.com/kubernetes-csi/external-snapshotter/client/v8/apis/volumegroupsnapshot/v1beta1"
+	snapapi "github.com/kubernetes-csi/external-snapshotter/client/v8/apis/volumesnapshot/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type groupSnapshotterType string
+
+type ocsGroupSnapshotClass struct{}
+
+const (
+	rbdGroupSnapshotter    groupSnapshotterType = "rbd"
+	cephfsGroupSnapshotter groupSnapshotterType = "cephfs"
+)
+
+const (
+	groupSnapshotterSecretNameKey      = "csi.storage.k8s.io/group-snapshotter-secret-name"
+	groupSnapshotterSecretNamespaceKey = "csi.storage.k8s.io/group-snapshotter-secret-namespace"
+)
+
+type GroupSnapshotClassConfiguration struct {
+	groupSnapshotClass *groupsnapapi.VolumeGroupSnapshotClass
+	reconcileStrategy  ReconcileStrategy
+	disable            bool
+}
+
+func newVolumeGroupSnapshotClass(instance *ocsv1.StorageCluster, groupSnaphotterType groupSnapshotterType) *groupsnapapi.VolumeGroupSnapshotClass {
```

Comment:
`groupSnaphotterType` is misspelled have `type`. If you have done something like this at other places also in response to my below comment pls change.

https://github.com/red-hat-storage/ocs-operator/pull/2859#discussion_r1905122678